packages/astro/src/core/dev/index.ts

```diff
@@ -24,7 +24,15 @@ export default async function dev(config: AstroConfig, options: DevOptions = { l
 		exclude: 'window document',
 	});
 	// start the server
-	const viteUserConfig = vite.mergeConfig({ mode: 'development' }, config.vite || {});
+	const viteUserConfig = vite.mergeConfig(
+		{
+			mode: 'development',
+			server: {
+				host: config.devOptions.hostname,
```

Comment:
In JS, a host would include the port. Did `vite` just decide otherwise in their API? If so, does this cover us from inheriting it?

Reply:
I'm not sure their reasoning, but I've confirmed that this value is meant to be the "hostname" and that `port` is kept entirely separate